src/manager/node.c

```diff
@@ -170,6 +170,20 @@ static int node_match_job_done(UNUSED sd_bus_message *m, UNUSED void *userdata,
         return 1;
 }
 
+static int node_match_heartbeat(UNUSED sd_bus_message *m, UNUSED void *userdata, UNUSED sd_bus_error *error) {
+        char *node_name = NULL;
+
+        int r = sd_bus_message_read(m, "s", &node_name);
+        if (r < 0 || node_name == NULL) {
+                fprintf(stderr, "Error reading heartbeat: %m\n");
```

Comment:
strings in dbus can't be null, that would be an error, so no need for the null check

Reply:
@alexlarsson can it be empty? Is there a point in checking `strlen`?

Reply:
Empty strings are valid dbus values, but I don't know why it would matter here. We basically ignore the passed in string anyway.

Reply:
I removed the null check.  Also, I added a print for just the first heartbeat received.